### ISSUE TITLE
[Global] Inconsistent checkbox styling

### ISSUE THREAD

**Issue by svinkle** [Shopify/dawn#1571]
## Issue summary

The checkbox border styling can sometimes be inconsistent, specifically in Safari. Depending on zoom level, borders may become thinner or thicker, or be missing entirely.

Discovered while running usability testing with screen magnification user.

<details>
  <summary>Screenshots</summary>
  
  <img width="438" alt="Dawn facet filter checkboxes. Box visual includes thin boarders, thick border, and missing border along the bottom." src="https://user-images.githubusercontent.com/1392632/161289754-ac69e444-090e-452a-9812-ded4f8c92fb2.png">

</details>

<!-- ### Current code

#### HTML

```html

``` -->

## Steps to reproduce

1. Go to any Dawn demo store with Safari
2. Use <kbd>Cmd</kbd> + <kbd>+</kbd> or <kbd>-</kbd> to zoom the viewport in/out
3. Enable [macOS Zoom](https://support.apple.com/en-ca/HT210978) and adjust the zoom level

## Behavior

### Expected

- Checkbox styles to be consistent.

### Actual

- Checkbox borders are inconsistent.

## Recommendation

Swap the SVG element for custom styled `input type="checkbox"` elements.

Refer to the article [One last time: custom styling radio buttons and checkboxes](https://www.scottohara.me/blog/2021/09/24/custom-radio-checkbox-again.html) by @scottaohara for details on how to accomplish this with accessibility in mind.

<!-- ### Recommended code

#### HTML

```html

``` -->

<!-- #### Diff

```diff
- line removed
+ line added
``` -->

## Specifications

- **Dawn version:** 4.0
- **Browser:** Safari
- **Component:** Global
- **WCAG Principal:** Perceivable
- **WCAG SC:** [1.3.3: Sensory Characteristics](https://www.w3.org/WAI/WCAG21/Understanding/sensory-characteristics.html)
- **Severity:** Low
- **Effort:** Medium
